packages/client/hooks/useUnusedRecords.ts

```diff
@@ -0,0 +1,18 @@
+import {useMemo} from 'react'
+
+const useUnusedRecords = (
+  records: ReadonlyArray<{readonly node: {readonly id: string}}>,
+  usedRecordIds: Set<string>
+): [string[], boolean | null] => {
+  return useMemo(() => {
+    const unusedRecords = [] as string[]
+    records.forEach(({node}) => {
+      if (!usedRecordIds.has(node.id)) unusedRecords.push(node.id)
+    })
+    const allSelected =
```

Comment:
i changed the naming to `allSelected` from `selectAll`. `selectAll` made me think the meaning was "should we select all right now or not?" `allSelected` makes it clearer that the meaning is whether all items have already all been selected.